src/fidesctl/cli/utils.py

```diff
@@ -295,8 +290,48 @@ def handle_aws_credentials_options(
             credentials_config=fides_config.credentials,
             credentials_id=credentials_id,
         )
-        if not aws_config:
+        _validate_credentials_id_exists(credentials_id, aws_config)
+    return aws_config
+
+
+def handle_bigquery_config_options(
+    fides_config: FidesctlConfig,
+    dataset: str,
+    # keyfile_credentials: KeyfileCreds,
+    keyfile_path: str,
+    credentials_id: str,
+) -> Optional[BigQueryConfig]:
+    """
+    Handles the connections options for passing a keyfile, dictionary, or credentials-id.
+    """
+    bigquery_config = None
+    if keyfile_path:
```

Comment:
```suggestion
    if keyfile_path and credentials_id:
```
i think this would work?

Reply:
Not necessarily for today, but I was thinking about maybe making a common pattern out of something similar to the below?
```
config_option_count = sum( config_option is not None for config_option in [keyfile_path, credentials_id])
if config_option_count != 1:
    raise click.UsageError(
        "Illegal usage: One and only one config option must be supplied."
    )
```

Reply:
in this case I think `any` is a better choice, for example:

```
In [1]: foo = [None, "foo"]

In [2]: if any(foo):
   ...:     print("It has a valid value!")
   ...:
It has a valid value!
```

Reply:
Would `any` be able to support the need for one (and only one) connection method passed via the cli? Or just validate that at least one is there?

Reply:
oh I see what you're doing, in that case your solution is the right one 💯 

only other nit here is that i think it could be:

```python
if keyfile_path and credentials_id:
   some_code
elif keyfile_path:
  some_code
elif credentials_id:
  some_code
else:
  print("No credentials provided")
```

just food for thought and not a blocker here

Reply:
Excellent call 👍🏽